client/parties/parties-reducer.ts

```diff
@@ -0,0 +1,94 @@
+import { List, Map, Record } from 'immutable'
+import { PartyUser } from '../../common/parties'
+import keyedReducer from '../reducers/keyed-reducer'
+
+export class PartyUserRecord
+  extends Record({
+    id: 0,
+    name: '',
+  })
+  implements PartyUser {}
+
+export class PartyRecord extends Record({
+  id: '',
+  invites: Map<number, PartyUser>(),
+  members: Map<number, PartyUser>(),
+  leader: new PartyUserRecord(),
+  messages: List(),
+}) {}
+
+export class PartyInvite extends Record({
+  partyId: '',
+  from: new PartyUserRecord(),
+}) {}
+
+export class PartiesState extends Record({
+  party: new PartyRecord(),
+  invites: Map<string, typeof PartyInvite>(),
```

Comment:
Yeah, I really don't get the purpose of this structure. If we're using notifications to accept party invites, the notification should contain the info about the party we've been invited to and are accepting. This is duplicated information that we don't have a use for, and is likely to fall out of sync